admin_guide/ui_extensions.adoc

```diff
@@ -0,0 +1,154 @@
+= Web Console Extensions
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+
+toc::[]
+
+== Loading Custom Scripts and Stylesheets
+Administrators can customize the
+link:../architecture/infrastructure_components/web_console.html[web console]
+using extensions, which let you run scripts and load custom stylesheets when
+the web console loads. You can change the look and feel of nearly any aspect of
+the user interface in this way.
+
+To add scripts and stylesheets, edit the
+link:master_node_configuration.html[master configuration]. The files must exist
+on the Asset Server and are added with the following options:
+
+----
+assetConfig:
+  extensionScripts:
+    - extensions/script1.js
+    - extensions/script2.js
+  extensionStylesheets:
+    - extensions/stylesheet1.css
+    - extensions/stylesheet2.css
+----
+
+Relative paths are resolved relative to the master configuration file. To pick
+up configuration changes, restart the server.
+
+Generally, the web console only reads custom scripts and stylesheets once on
+startup. The web console has an extension development mode, however, that lets
+you test changes to these files without having to restart after every edit.
+This is enabled through the following master configuration option:
+
+----
+assetConfig:
+  extensionDevelopment: true
+----
+
+When set, the web console will reload any changes to existing extension script
+or stylesheet files when you refresh the page in your browser. You still must
+restart the server when adding new extension stylesheets or scripts, however.
+This setting is only recommended for testing changes and not for production.
+
+=== Customizing the Banner Logo
+
+To customize the logo in the web console banner, add the following style to a
+CSS file, for example `theme.css`.
+
+----
+#brand-logo {
+  background-image: url("https://www.example.com/images/logo.png");
+  width: 160px;
+  height: 10px;
+}
+----
+
+Replace the example.com URL with a URL to an actual image and adjust the width
+and height. Copy `theme.css` to the same directory as the master configuration
+file, then add this option:
+
+----
+assetConfig:
+  extensionStylesheets:
+    - theme.css
+----
+
+Restart the server to the see the new logo. The recommended height for a
+custom logo is 10px.
+
+== Serving Static Files
+
+You can serve other files form the Asset Server as well. For instance, you
+might want to make the `oc` command-line executable available for download from
+the web console or add images to use in a custom stylesheet.
+
+Add the directory with the files you want using the following configuration option:
+
+----
+assetConfig:
+  extensions:
+    - name: images
+      sourceDirectory: extensions/my_images
```

Comment:
make this a more obvious example dir to avoid people correlating this dir with the URL